### ISSUE TITLE
Non blocking errors when filename contains [1]

### ISSUE THREAD

**Issue by OskarKlintrot** [kpfaulkner/azurecopy#31]
Command
```
> ./azurecopy.exe -i https://mybucket.s3.amazonaws.com/myblob -o https://myaccount.blob.core.windows.net/mycontainer -blobcopy
```
gives
```
StartCopyFromBlob error msg Fjärrservern returnerade ett fel: (403) Förbjuden.
StartCopyFromBlob error stack    vid Microsoft.WindowsAzure.Storage.Core.Executor.Executor.ExecuteSync[T](RESTCommand`1 cmd, IRetryPolicy policy, OperationContext operationContext)
   vid azurecopy.AzureBlobCopyHandler.StartCopy(BasicBlobContainer origBlob, String DestinationUrl, DestinationBlobType destBlobType, Boolean skipIfExists)
Unable to start copying pexels-photo-303024[1].jpeg
```

```
Fjärrservern returnerade ett fel: (403) Förbjuden.
```

would probably translate to

```
Remote server returned an error: (403) Forbidden.
```

This is the only file that we couldn't copy and the only file containing `[1]` (both in the filename and path, which is technically the same). 

ps. Thanks for an awesome program, we owe you a beer! ds.

**Comment by kpfaulkner** [comment]
Hi

Just to check, in the example the blob is called "myblob" but in reality it
has the characters [1] in it?  ie something like myblob[1] ?

Correct?

if so, I'll try and repro it here and figure out a fix. I bet those []'s
will cause an issue.

Cheers

Ken

**Comment by OskarKlintrot** [comment]
Thank God for the history in Powershell! No, in the example I had just copy/pasted the example from the readme. This is the "actual" command from my Powershell history:
```
> ./azurecopy.exe -i https://mybucket.s3.amazonaws.com -o https://myaccount.blob.core.windows.net/mycontainer -blobcopy
```
So it tried to copy everything from S3 to Blob Storage but failed on that file with a `[1]` in it. I haven't looked in the source but I guess it assumed it was an array and tried to iterate over it or something like that.

Sorry for the confusion!

**Comment by kpfaulkner** [comment]
Hi

Ok, I'll repro this later today/tonight and sort out a fix.

Cheers

Ken

**Comment by kpfaulkner** [comment]
Hi

I just tried it with a blob called test1[1], copying from S3 to Azure and
it worked fine. No issues at all.
Can you give me the entire blobname? Maybe there is something else that
combines with the [1] that causes the problem.

Thanks

Ken

**Comment by OskarKlintrot** [comment]
The name should be:
```
images/pexels-photo-303024[1]/original/pexels-photo-303024[1].jpeg
```

There where also a few that hade namings like this:

```
images/pexels-photo-303024[1]/w_500-h_300/pexels-photo-303024[1].jpeg
```

I just got a bunch of the error message inserted in my first post so unfortunately I have no clue which one caused the issue.

**Comment by kpfaulkner** [comment]
Cheers, I'll give that a go.

**Comment by OskarKlintrot** [comment]
Great, thanks!

**Comment by kpfaulkner** [comment]
hmmmm still no luck for me. It's working fine.

Can you copy/paste some of the output from when you ran it?

**Comment by OskarKlintrot** [comment]
Sorry, I don't have more than that. The blob storage container is now in production so I don't really want to re-run the command...

**Comment by kpfaulkner** [comment]
fair enough. I'll try some alternatives and try and see if I can get any
similar errors.

Thanks

Ken

**Comment by kpfaulkner** [comment]
Ahhh able to repro! I missed out the -blobcopy flag. Ok, will investigate further.

**Comment by kpfaulkner** [comment]
Have checked if its the S3 or Azure side.  It's definitely Azure, but unsure if there is anything I can do about it. The shared URL that S3 makes is definitely valid (ie I can copy it to a browser and can download manually). I'm still investigating if there is anything I can do on the Azure side. I don't believe its the Azure name containing [1]... but it's more the fact when we tell Azure to copy the file, it dies.

Honestly unsure if there is anything I can do about it, but will keep investigating.

Cheers

Ken
